### ISSUE TITLE
Stuck in CactusReferenceRecursion3 on small toy example

### ISSUE THREAD

**Issue by jmonlong** [ComparativeGenomicsToolkit/cactus#182]
Hi, I'm trying to test cactus on a small dataset (2 sequences: one of 100kbp, the other is the same but with a deletion in the middle). I'm running cactus locally and the workflow progresses quite quickly but then seems to get stuck in job "CactusReferenceRecursion3" (see [cactus-test.log.txt](https://github.com/ComparativeGenomicsToolkit/cactus/files/4366001/cactus-test.log.txt)). I don't see much processor activity in *top*.

I'm not sure if it's a problem with the input data or something similar to the other issues I have seen about "Reissued any over long jobs".

I ran the following using [these input files (zip file)](https://github.com/ComparativeGenomicsToolkit/cactus/files/4365988/cactus.100kbp.simpledel.zip):

```
cactus jobStore test.txt test.hal
```

I tried `--restart`ing the workflow but it gets stuck at the same step.

Do you have a suggestion to debug or run differently?
Thanks

**Comment by glennhickey** [comment]
I just ran it on my desktop `cactus jobStore test.txt test.hal --realTimeLogging --logInfo --binariesMode local` and it worked

```
Finished toil run successfully.
Stopping real-time logging server.
Joining real-time logging server thread.
Successfully deleted the job store: FileJobStore(/home/hickey/dev/work/jean-cactus/jobStore)
Cactus has finished after 656.0319809019566 seconds
```

I will try your command (which will pull binares from docker) on the latest master branch and see what happens.

**Comment by glennhickey** [comment]
It also goes through for me on a fresh clone
```
git clone https://github.com/ComparativeGenomicsToolkit/cactus.git --recursive cactus-fresh
cd cactus-fresh
virtualenv -p python3.6 fresh-venv
. fresh-venv/bin/activate
pip install -U .
cactus jobStore test.txt test.hal
```

I'd suggest `--realTimeLogging --logDebug` to get some more error messages.  One cause of an infinite hang is toil not finding enough free ram/cpu/disk to schedule the next job, and this is only made apparent with `--logDebug`.  If you're running from inside a docker image, you'll need `--binariesMode local` I think.  Is this on a system I can log into?

**Comment by jmonlong** [comment]
Thanks for checking. It might be my laptop which doesn't have much memory. I'll try with the logging parameters or on a bigger machine. I was using it with `--binariesMode local` within a docker image, I forgot to include this part in the original message.

**Comment by jmonlong** [comment]
I'm still getting the problem and I'm wondering if it has to do with the fact I'm using docker. I tried running on a machine with more memory (*plaza*) and got the same kind of problem, although at a different step. 

It get stucks with the following log-info repeating:

```
2020-03-22 22:51:44.795018: Running the command: "ktremotemgr get -port 15866 -host 172.17.0.13 TERMINATE"
```

I created the docker container from the repo:

```
git clone https://github.com/ComparativeGenomicsToolkit/cactus.git --recursive cactus
cd cactus
docker build -t jmonlong-cactus .
```
And then ran:

```
docker run -it -v `pwd`:/app -u `id -u $USER` -w /app jmonlong-cactus cactus cactus/jobStore_hsvlr-100bp_2_ins cactus/hsvlr-100bp_2_ins.cactus.inputs cactus/hsvlr-100bp_2_ins-cactus.hal --binariesMode local --maxCores 1 --logFile cactus/hsvlr-100bp_2_ins-cactus.log --realTimeLogging --logInfo
```

from `/public/groups/vg/jmonlong/align_sv_sandbox` on *courtyard/plaza* (if you want to try it out @glennhickey). For example the full log is at `/public/groups/vg/jmonlong/align_sv_sandbox/cactus/hsvlr-100bp_2_ins-cactus.log`

**Comment by glennhickey** [comment]
Cactus doesn't work with `--maxCores 1` and it really needs an error message to let people know.

**Comment by glennhickey** [comment]
Not even sure it works with `--maxCores 2` actually.  3 or more should be fine though!

**Comment by jmonlong** [comment]
Thanks for the info, it did work with `--maxCores 3` on courtyard!

I still get repeating `Running the command: "ktremotemgr get -port 26317 -host 172.17.0.2 TERMINATE"` messages on my laptop but that might be because of limited memory as you suggested.

**TL;DR of this issue: even for a small toy example, Cactus needs a minimum amount of free memory (>4Gb?) and cores (>2).**

**Comment by glennhickey** [comment]
Yeah, while it technically doesn't need those resources, there are some minimums baked into the system that would need to be hacked to run with less.  

There's also a constant factor in terms of runtime  that makes it take 10 minutes even on your tiny example.
